internal/signutil/signutil.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2023 Stefan Prodan
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package signutil
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+
+	"github.com/go-logr/logr"
+)
+
+// Sign signs an image (`imageRef`) using a cosign private key (`keyRef`)
+func Sign(log logr.Logger, imageRef string, keyRef string) error {
+	cosignExecutable, err := exec.LookPath("cosign")
+	if err != nil {
+		return fmt.Errorf("executing cosign failed: %w", err)
+	}
+
+	cosignCmd := exec.Command(cosignExecutable, []string{"sign"}...)
+	cosignCmd.Env = os.Environ()
+
+	// if key is empty, use keyless mode(experimental)
```

Comment:
Keyless is no longer experimental since Cosign v2. Do we really need `COSIGN_EXPERIMENTAL=true`?

Reply:
I added it for older Cosign versions compatibility, but if it's not that important, I can remove it

Reply:
The old one v1, didn’t had the `—yes` flag so it would fail anyway

Reply:
Yep, just checked it, I guess I just remove it then